src/main/java/org/opensearch/geospatial/ip2geo/jobscheduler/Datasource.java

```diff
@@ -322,19 +331,43 @@ public boolean isExpired() {
         return Instant.now().isAfter(lastCheckedAt.plus(database.validForInDays, ChronoUnit.DAYS));
     }
 
-    public void setDatabase(final DatasourceManifest datasourceManifest, final String[] fields) {
+    /**
+     * Set database attributes with given input
+     *
+     * @param datasourceManifest the datasource manifest
+     * @param fields the fields
+     */
+    public void setDatabase(final DatasourceManifest datasourceManifest, final List<String> fields) {
         this.database.setProvider(datasourceManifest.getProvider());
         this.database.setMd5Hash(datasourceManifest.getMd5Hash());
         this.database.setUpdatedAt(Instant.ofEpochMilli(datasourceManifest.getUpdatedAt()));
-        this.database.setValidForInDays(database.validForInDays);
-        this.database.setFields(Arrays.asList(fields));
+        this.database.setValidForInDays(datasourceManifest.getValidForInDays());
+        this.database.setFields(fields);
+    }
+
+    /**
+     * Checks if the database fields are compatible with the given set of fields.
+     *
+     * @param fields The set of input fields to check for compatibility.
+     * @return true if the database fields are compatible with the given input fields, false otherwise.
+     */
+    public boolean isCompatible(final List<String> fields) {
+        Set<String> fieldsSet = new HashSet<>(fields);
+        for (String field : database.fields) {
```

Comment:
you can also use size to decide if they are not compatible before checking all fields.